lib/util/package-rules/index.ts

```diff
@@ -73,25 +73,28 @@ export function applyPackageRules<T extends PackageRuleInputConfig>(
     // This rule is considered matched if there was at least one positive match and no negative matches
     if (matchesRule(config, packageRule)) {
       // Package rule config overrides any existing config
-      const toApply = { ...packageRule };
+      const toApply = removeMatchers({ ...packageRule });
       if (config.groupSlug && packageRule.groupName && !packageRule.groupSlug) {
         // Need to apply groupSlug otherwise the existing one will take precedence
         toApply.groupSlug = slugify(packageRule.groupName, {
           lower: true,
         });
       }
       config = mergeChildConfig(config, toApply);
-      delete config.matchPackageNames;
-      delete config.matchPackagePatterns;
-      delete config.matchPackagePrefixes;
-      delete config.excludePackageNames;
-      delete config.excludePackagePatterns;
-      delete config.excludePackagePrefixes;
-      delete config.matchDepTypes;
-      delete config.matchCurrentValue;
-      delete config.matchCurrentVersion;
-      delete config.matchCurrentAge;
     }
   }
   return config;
 }
+
+function removeMatchers(
```

Comment:
Is it better to create a new object like this or just to iterate through the existing one and delete applicable keys?

Reply:
I don't see any issue in both. It's a matter of choice since the performance difference isn't much.

Reply:
In that case please refactor to delete

Reply:
Generally its bad practice to delete the keys while looping over an object, but it seems its okay in javasript.
[Reference ](https://stackoverflow.com/questions/3463048/is-it-safe-to-delete-an-object-property-while-iterating-over-them)

Reply:
Object.keys is evaluated before the loop, so no issue

Reply:
It's already cloned due to mergeChildConfig

Reply:
ok, so delete is the only option here